tests/generic_config_updater/test_dynamic_acl.py

```diff
@@ -0,0 +1,1251 @@
+                # we record the name of the portchannel interface here, as this is the actual
+                # interface the DHCP relay will listen on.
+                iface_is_portchannel_member = False
+                for portchannel_name, portchannel_info_dict in list(mg_facts['minigraph_portchannels'].items()):
+                    if 'members' in portchannel_info_dict and iface_name in portchannel_info_dict['members']:
+                        iface_is_portchannel_member = True
+                        if portchannel_name not in uplink_interfaces:
+                            uplink_interfaces.append(portchannel_name)
+                        break
+                # If the uplink's physical interface is not a member of a portchannel,
+                # add it to our uplink interfaces list
+                if not iface_is_portchannel_member:
+                    uplink_interfaces.append(iface_name)
+                uplink_port_indices.append(mg_facts['minigraph_ptf_indices'][iface_name])
+
+    # Obtain MAC address of an uplink interface because vlan mac may be different than that of physical interfaces
+    res = rand_selected_dut.shell('cat /sys/class/net/{}/address'.format(uplink_interfaces[0]))
+    uplink_mac = res['stdout']
+
+    """ update_payload method which is automatically called in ptfadapter on .send() or any .verify_packet() method
+        is bugged for dhcp_discover packets.  Need to override it to do nothing."""
+
+    def new_update_payload(pkt):
+        return pkt
+
+    ptfadapter.update_payload = new_update_payload
+
+    setup_information = {
+        "blocked_src_port_name": block_src_port,
+        "blocked_src_port_indice": block_src_port_indice,
+        "blocked_src_port_alias": block_src_port_alias,
+        "unblocked_src_port_indice": unblocked_src_port_indice,
+        "scale_port_names": scale_ports,
+        "scale_port_indices": scale_ports_indices,
+        "scale_dest_ips": scale_dest_ips,
+        "dst_port_indices": dst_port_indices,
+        "router_mac": router_mac,
+        "bind_ports": list_ports,
+        "dut_mac": dut_mac,
+        "vlan_ips": vlan_ips,
+        "is_dualtor": is_dualtor,
+        "switch_loopback_ip": switch_loopback_ip,
+        "ipv4_vlan_mac": v4_vlan_mac,
+        "uplink_mac": uplink_mac,
+    }
+
+    return setup_information
+
+
+@pytest.fixture(autouse=True)
+def setup_env(duthosts, rand_one_dut_hostname):
+    """
+    Setup/teardown fixture for acl config
+    Args:
+        duthosts: list of DUTs.
+        rand_selected_dut: The fixture returns a randomly selected DuT.
+    """
+    duthost = duthosts[rand_one_dut_hostname]
+    create_checkpoint(duthost)
+
+    yield
+
+    try:
+        logger.info("Rolled back to original checkpoint")
+        rollback_or_reload(duthost)
+    finally:
+        delete_checkpoint(duthost)
+
+
+@pytest.fixture
+def proxy_arp_enabled(rand_selected_dut, config_facts):
```

Comment:
Is proxy_arp required for the ARP test?

Reply:
The ARP test that I based my packets off of was a proxy_arp test which also required proxy_arp enabled within it, so I ported this over.  Before using it, I would sometimes see ARP fail sporadically on a DUT I had reserved

Reply:
For this particular ARP test, proxy_arp is probably not required because we configured IP address on the ptf port, and send ARP request to that specific port. Can you please remove this code and rerun the test and see if it can pass?
I don't suggest changing the proxy_arp setting because earlier I saw the setting triggered some other issues, which is fixed in another PR https://github.com/sonic-net/sonic-mgmt/pull/11977

Reply:
removed